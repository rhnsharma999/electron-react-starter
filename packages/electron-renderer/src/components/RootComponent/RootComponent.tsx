import * as React from "react"
// type checking seems to be broken for props in vscode, need to fix that.
class RootComponent extends React.Component {
  state = {
    nodeVersion: "",
    chromeVersion: "",
    electronVersion: "",
  }

  componentDidMount() {
    const nodeVersion = process.versions.node
    const chromeVersion = process.versions.chrome
    const electronVersion = process.versions.electron
    this.setState({ nodeVersion, chromeVersion, electronVersion })
  }
  render() {
    const { nodeVersion, chromeVersion, electronVersion } = this.state

    return (
      <div>
        <h3>Node = v${nodeVersion}</h3>
        <h3>Chrome = v${chromeVersion}</h3>
        <h3>Electron = v${electronVersion}</h3>
      </div>
    )
  }
}

export default RootComponent
