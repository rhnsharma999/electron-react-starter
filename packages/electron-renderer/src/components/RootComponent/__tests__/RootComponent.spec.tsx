import { shallow } from "enzyme"
import * as React from "react"
import RootComponent from ".."

describe("this is a sample test", () => {
  it("executes succesfully", () => {
    const wrapper = shallow(<RootComponent />)
    expect(wrapper).toMatchSnapshot()
  })
})
