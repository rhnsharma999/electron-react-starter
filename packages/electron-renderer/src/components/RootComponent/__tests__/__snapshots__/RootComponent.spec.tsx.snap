// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`this is a sample test executes succesfully 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Click to contact background script 
  </button>
</div>
`;
